### PR TITLE
Add cb bit rotations

### PR DESCRIPTION
**Pull request by ricfehr3**
Add all cb bit rotations. 

Name change shift methods in mod.rs to rotate. Plus bugfixes.

### CODE REVIEW

File: src/cpu/test.rs [hunk 1]
```diff
@@ -814,6 +831,79 @@ fn test_shift() {
     assert_eq!(cpu.get_flag(Flag::H), false);
     assert_eq!(cpu.get_flag(Flag::C), true);
 
+    cpu.set_reg(Register8Bit::B, 0x95);
```
Comment (toastbucket) [inline src/cpu/test.rs:834]: Can you add comments before each test to indicate which instruction its testing?

File: src/cpu/instruction.rs [hunk 2]
```diff
@@ -652,6 +660,46 @@ pub enum CbInstruction {
 impl CbInstruction {
     pub fn from_byte(byte: u8) -> Option<CbInstruction> {
         match byte {
+            // rlc register 8 bit
```
Comment (toastbucket) [inline src/cpu/instruction.rs:663]: stick with the naming convention from the manual for these comments

File: src/cpu/test.rs [hunk 3]
```diff
@@ -814,6 +831,79 @@ fn test_shift() {
...
+    cpu.set_reg(Register8Bit::B, 0x85);
+    cpu.set_flag(Flag::C, true);
+    cpu.step();
+    assert_eq!(cpu.get_reg(Register8Bit::B), 0x0b);
+    assert_eq!(cpu.get_flag(Flag::Z), false);
+    assert_eq!(cpu.get_flag(Flag::N), false);
+    assert_eq!(cpu.get_flag(Flag::H), false);
+    assert_eq!(cpu.get_flag(Flag::C), true);
+
+    cpu.set_reg(Register8Bit::B, 0x81);
```
Comment (toastbucket) [inline src/cpu/test.rs:852]: Let's use the test cases from the manual for all of these to be safe.

**Review by toastbucket — CHANGES_REQUESTED** [review]

File: src/cpu/instruction.rs (hunk 2 above)
Comment (ricfehr3) [reply src/cpu/instruction.rs:663]: You want like, "rlc n"?

File: src/cpu/test.rs (hunk 1 above)
Comment (ricfehr3) [reply src/cpu/test.rs:834]: https://tenor.com/view/bruh-gpo-bad-trade-nty-bro-gif-24142673

File: src/cpu/test.rs (hunk 3 above)
Comment (ricfehr3) [reply src/cpu/test.rs:852]: These are your tests for rla,rlca,rra,rrca copies lol. There are no tests lined out for other rotations.

File: src/cpu/test.rs [hunk 4]
```diff
@@ -814,6 +831,79 @@ fn test_shift() {
     assert_eq!(cpu.get_flag(Flag::H), false);
     assert_eq!(cpu.get_flag(Flag::C), true);
 
+    cpu.set_reg(Register8Bit::B, 0x95);
+    cpu.set_flag(Flag::C, true);
```
[the remark this answers is not in the document]
Comment (ricfehr3) [reply src/cpu/test.rs:835]: You put me in a tough spot, buddy. This follows the tests in the manual. I agree with you, but what about the manual :1st_place_medal:

[the remark this answers is not in the document]
Comment (toastbucket) [reply src/cpu/test.rs:840]: U lookin at the wrong tests daawwwwgggg

File: src/cpu/test.rs (hunk 3 above)
Comment (toastbucket) [reply src/cpu/test.rs:859]: check again my duuuuuuude

**Pull request merged**